akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -88,17 +88,35 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
 
       override def doPush(elem: ElementAndTrigger): Unit =
         elem.element match {
-          case d @ DataFrame(streamId, _, pl) ⇒
-            if (pl.size <= totalOutboundWindowLeft && pl.size <= windowLeftFor(streamId)) {
+          case d @ DataFrame(streamId, _, payload) ⇒
+            val sendSize = totalOutboundWindowLeft min windowLeftFor(streamId) min payload.size
+
+            def send(elem: ElementAndTrigger): Unit = {
```

Comment:
Ah I see, this is handling window size, whereas the renderer was handling frame size limits. 
Good as is!